anomalib/utils/callbacks/visualizer_callback.py

```diff
@@ -133,25 +139,26 @@ def on_test_batch_end(
             pred_mask = compute_mask(anomaly_map, threshold)
             vis_img = mark_boundaries(image, pred_mask, color=(1, 0, 0), mode="thick")
 
-            num_cols = 6 if self.task == "segmentation" else 5
-            visualizer = Visualizer(num_rows=1, num_cols=num_cols, figure_size=(12, 3))
-            visualizer.add_image(image=image, title="Image")
-
-            if "mask" in outputs:
-                true_mask = outputs["mask"][i].cpu().numpy() * 255
-                visualizer.add_image(image=true_mask, color_map="gray", title="Ground Truth")
-
-            visualizer.add_image(image=heat_map, title="Predicted Heat Map")
-            visualizer.add_image(image=pred_mask, color_map="gray", title="Predicted Mask")
-            visualizer.add_image(image=vis_img, title="Segmentation Result")
-
-            image_classified = visualizer.add_text(
-                image=image,
-                text=f"""Pred: { "anomalous" if pred_score > threshold else "normal"}({pred_score:.3f}) \n
-                GT: {"anomalous" if bool(gt_label) else "normal"}""",
-            )
-            visualizer.add_image(image=image_classified, title="Classified Image")
+            visualizer = Visualizer()
+
+            if self.task == "segmentation":
```

Comment:
It's because we only want to show masks when the task type is segmentation. Otherwise, when the task type is classification, we show the classification style visualization.